htdocs/js/ui/menus.js

```diff
@@ -58,6 +58,21 @@ RCloud.UI.menu = (function() {
                     var item = extension_.get(menu_item);
                     if(!item || !item.$li)
                         throw new Error('menu disable fail on ' + menu_item);
+
+                    if(!enable) {
+                        item.enabled_action = item.action;
+                        item.action = function() {}    
+                        if(item.disabled_reason) {
+                            item.$li.find('a').attr('title', item.disabled_reason);
+                        }                
+                    } else {
+                        if(item.enabled_action) {
+                            item.action = item.enabled_action;
+                            delete item.enabled_action;
+                        }
+                        item.$li.find('a').removeAttr('title');
+                    }
+
```

Comment:
Whoops, if it gets disabled twice (e.g. two readonly notebooks opened in a row), then the `action` is completely lost. Replacing with a disabled flag which gets checked before `.action()` is invoked.